protocol/proofs/generic-withdrawal-proofs.md

```diff
@@ -0,0 +1,98 @@
+# Super Root Withdrawal Proofs
+
+## Purpose
+
+Proposes a modification to the `OptimismPortal` that would add a new overload of
+`proveWithdrawalTransaction` that would allow users to prove against Super Roots.
+
+## Context
+
+Interop introduces the concept of the "Super Root" - a collection of Output Roots from all of the
+different chains in the interop set. The interop fault proof system argues over a super root rather
+than an individual Output Root. The `OptimismPortal` contract only supports proving withdrawals
+against a specific Output Root and will need to be modified to support the Super Root.
+
+## Problem Statement
+
+We need to modify the `OptimismPortal` contract to support proving withdrawals against a Super
+Root. Users would not be able to correctly execute withdrawals without this modification. This
+means that *some* solution to the problem is a basic requirement for interop.
+
+## Proposed Solution
+
+We propose to allow users to prove against Super Roots by adding a new overload of
+`proveWithdrawalTransaction` that allows a user to prove the existence of an Output Root inside of
+the Super Root before performing the standard withdrawal validation inside of the Output Root.
+
+We also specifically propose that this be implemented on the *base* `OptimismPortal` instead of
+introducing the code to the `OptimismPortalInterop` portal. Doing so means that we are including
+extra code on the `OptimismPortal` that is technically unnecessary until interop ships but also
+means that we do not need to ship any additional changes to the `OptimismPortal` before interop.
+
+We could expose a new public variable `superRootsActive` that would toggle this function on and off
+and would also act as a signal for client-side tooling to automatically switch between the two
+available functions. When `superRootsActive` is `false`, the version of
+`proveWithdrawalTransaction` for Super Roots would revert. When `superRootsActive` is `true`, the
+legacy version of the function would revert.
+
+## Alternatives Considered
+
+### Using an Inherited Contract
+
+We could implement this function inside of an inherited contract instead of putting it in the base
+contract. However, this would mean performing two sets of upgrades to the `OptimismPortal` before
+interop ships on an already tight timeline.
+
+## Considerations
+
+### Unused Code in Production
+
+This proposal means we are shipping code that cannot yet be used into production. We believe this
+trade-off is worthwhile because it means that we would not have to ship the code later on.
```

Comment:
This is a bit precedent-setting for us, as we typically do not ship dead code to production. It's likely we start doing things like this more often to help manage merging of in-flight features to develop, so it would be nice to have clear guidelines for implementation and testing of dead/feature-flagged code

Reply:
Good callout! Where would be the right place for this? Notion? Github?

Reply:
I would suggest a new design doc to supersede https://github.com/ethereum-optimism/design-docs/blob/main/smart-contract-feature-development.md, with the result being an update to our style guide. So perhaps the design review is really just a review of a proposed PR to the style guide

Reply:
Ok I'll make a note on my end that this should get updated but won't block this design doc on it